scripts/rules.go

```diff
@@ -134,6 +134,42 @@ func databaseImport(m dsl.Matcher) {
 		Where(m.File().PkgPath.Matches("github.com/coder/coder/v2/codersdk"))
 }
 
+// publishInTransaction detects calls to Publish inside database transactions
+// which can lead to connection starvation.
+//
+//nolint:unused,deadcode,varnamelen
+func publishInTransaction(m dsl.Matcher) {
+	m.Import("github.com/coder/coder/v2/coderd/database/pubsub")
+
+	// Match direct calls to the Publish method of a pubsub instance inside InTx
+	m.Match(`
+		$x.InTx(func($y) error {
```

Comment:
A nit, but we don't care what these arguments are, so we do not need to name them.

```suggestion
		$_.InTx(func($_) error {
```

Reply:
Hhmm, apparently the arg for `func()` needs an identifier, otherwise the rule fails to work.
See 6e681d599

Reply:
TIL :+1: